### PR TITLE
homework done

### PR DESCRIPTION
**Pull request by islam-fawzy25**

### CODE REVIEW

File: databases/week2/part4/part4.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ id int unsigned auto_increment primary key,
+ name varchar(50),
+ email varchar(50)
+ );
+ 
+ CREATE TABLE developers (
+ id int unsigned auto_increment primary key ,
+ name varchar (50),
+ email varchar(50),
+ leader_id int unsigned ,
```
Comment (flemeur) [inline databases/week2/part4/part4.sql:25]: Would it maybe make more sense to have the `leader_id` foreign key on the teams table instead?

**Review by flemeur — COMMENTED** [review]
Hi @islam-fawzy25 Nice Work :+1: Here are my comments

File: databases/week2/part1.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+-- Add a task
+insert into task (title, description, created, updated, due_date, status_id)
+value ('new task','new task description',now(),now(),now(),1);
+
+-- Change the title of a task
+update task set title ='title updated'
+where id =36;
+
+-- Change a task due date
+update task set due_date = DATE("2021-06-15 09:34:21")
```
[the remark this answers is not in the document]
Comment (islam-fawzy25) [reply databases/week2/part1.sql:14]: > You don't need to use the date function here. The `due_date` is a DATETIME type so it makes sense to also include the time part. `'2021-06-15 09:34:21'`

yes , it makes sense. good point thanks

File: databases/week2/part2.sql [hunk 3]
```diff
@@ -0,0 +1,45 @@
+create database school_db ;
+use school_db;
+-- Class: with the columns: id, name, begins (date), ends (date)
+create table class (
+id int unsigned auto_increment primary key,
+name  varchar(255) not null unique,
```
[the remark this answers is not in the document]
Comment (islam-fawzy25) [reply databases/week2/part2.sql:6]: > Nice touch that you added a unique constraint on the name column

Thanks 🥰

File: databases/week2/part2.sql [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+first solution
+ALTER TABLE SomeTable
+        ADD SomeCol Bit NULL --Or NOT NULL.
+ CONSTRAINT D_SomeTable_SomeCol --When Omitted a Default-Constraint Name is autogenerated.
+    DEFAULT (0)--Optional Default-Constraint.
+WITH VALUES --Add if Column is Nullable and you want the Default Value for Existing Records.
+*/
+-- second solution : creat status table ,then alter column to class table 
+ALTER TABLE class 
+Drop status ;
```
[the remark this answers is not in the document]
Comment (islam-fawzy25) [reply databases/week2/part2.sql:39]: > Not needed

sorry , i was checking and forgot to remove that

File: databases/week2/part2.sql [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+Add a new column to the class table named status which can only have the
+ following values: not-started, ongoing, finished (hint: enumerations).
+*/
+/*
+first solution
+ALTER TABLE SomeTable
+        ADD SomeCol Bit NULL --Or NOT NULL.
+ CONSTRAINT D_SomeTable_SomeCol --When Omitted a Default-Constraint Name is autogenerated.
+    DEFAULT (0)--Optional Default-Constraint.
+WITH VALUES --Add if Column is Nullable and you want the Default Value for Existing Records.
```
[the remark this answers is not in the document]
Comment (islam-fawzy25) [reply databases/week2/part2.sql:35]: > Not sure what this is doing, but it is not what is meant in the exercise description

it was something from stack overflow . it did not help and I forgot to remove it

File: databases/week2/part3.sql [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ where user.name like 'Maryrose Meadows' and month(created) =09 ;
+ -- or :  created  between  '2017-09-01'  and '2017-09-30' ;
+ 
+ /*
+ Find how many tasks where created in each month, e.g.
+ how many tasks were created in october,
+ how many tasks were created in november, etc. (hint: use group by)
+ */
+select count(created), task.created from task
+group by month(created) ;
```
[the remark this answers is not in the document]
Comment (islam-fawzy25) [reply databases/week2/part3.sql:34]: > This last query gives me an error
> `Error in query (1055): Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'hyf_lesson2.task.created' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by`
> 
> Can you guess why? :)

it is weird, i got no Error 
but i think we can solve this Error by 
select count(task.id), task.created from task
group by month(created) ;

[the remark this answers is not in the document]
Comment (flemeur) [reply databases/week2/part3.sql:34]: I think the reason it works for you has to do with the configuration of mysql.

Basically the error means that you cannot select the created date while also grouping by the created month.

The correct query would have to be:
```
select count(id), month(created) from task
group by month(created);
```

Hope it makes sense :)

**Pull request merged**